app/src/test/java/com/example/househomey/TagFilterTest.java

```diff
@@ -0,0 +1,78 @@
+package com.example.househomey;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import com.example.househomey.filter.model.TagFilter;
+import com.example.househomey.tags.Tag;
+import com.google.firebase.Timestamp;
+import static org.junit.Assert.assertEquals;
+
+public class TagFilterTest {
+    private ArrayList<Item> itemList;
+
+    @Before
+    public void setUp() {
+        itemList = new ArrayList<>();
+        itemList.add(createItem("3", "Smartphone", "Electronics", BigDecimal.valueOf(799.99), "Samsung", "Galaxy", "789012", "Nice phone"));
+        itemList.add(createItem("1", "Laptop", "Electronics", BigDecimal.valueOf(999.99), "Dell", "XPS", "123456", "Good laptop"));
+        itemList.add(createItem("2", "Sofa", "Furniture", BigDecimal.valueOf(499.99), "", "", "", ""));
+    }
+
+    private Item createItem(String id, String description, String tagLabel, BigDecimal cost, String make, String model, String serialNumber, String comment) {
+        Map<String, Object> data = new HashMap<>();
+        data.put("description", description);
+        data.put("acquisitionDate", new Timestamp(new Date()));
+        data.put("cost", cost.toString());
+        data.put("make", make);
+        data.put("model", model);
+        data.put("serialNumber", serialNumber);
+        data.put("comment", comment);
+
+        Item item = new Item(id, data);
+        item.addTag(new Tag(tagLabel, data));
+        return item;
+    }
+
+    @Test
+    public void testNoTagSelected() {
+        Map<String, Boolean> tagSelectionMap = new HashMap<>();
+        tagSelectionMap.put("Electronics", false);
+        tagSelectionMap.put("Furniture", false);
+
+        TagFilter tagFilter = new TagFilter(tagSelectionMap);
```

Comment:
Your unit tests fail when I try and run locally because TagFilter's constructor no longer expects a Hashmap, it expects a Set.

Interestingly, the CI passed for unit tests though. So I took a look at the CI and found out they failed in there, but didn't notify us of the failure. The action continued and it was reported as success.
![image](https://github.com/CMPUT301F23T08/HouseHomey/assets/90405643/2729cf39-0614-45ba-90f9-e96da94cafbe)

There's an error in the Github Actions .yml file, can you please remove this line and commit it along with your unit test fixes. https://github.com/CMPUT301F23T08/HouseHomey/blob/daccdec7416829a452193fbfd606b38244f5d780/.github/workflows/android.yml#L44
It shouldn't be continuing on error, it should throw the error.

Reply:
Will do. Thanks for catching that.